### PR TITLE
PLJ-928 Bond value fix

### PR DESCRIPTION
**Pull request by pankgeorg**
There is a flag that's currently ignored : `is_first_value`. 

There is also a prophylactic (`προ-φύλαξις` - guard proactively) comment in [this line](https://github.com/fonsp/Pluto.jl/blob/54e6658ab7cb4b6a1c9572e9e4958f1f6f740604/frontend/components/Editor.js#L413) that talks about #275 - it looks that #275 doesn't re-appear.

@fons, @dralletje remember why the comment was for?

This fix does exactly the opposite of the comment, so let's discuss!

### CODE REVIEW

**Comment by dralletje** [comment]
It was meant as a reminder to use `is_first_value` again if #275 would be un-fixed because of Firebasey.
Something like #275 seems to be happening here though, so using `is_first_value` is exactly right.

However, with this simple prophylactic we prevent html binds from setting their initial value:
```julia
# ╔═╡ 20bd321a-9cec-4d6a-a618-cc8132fbc8c7
@bind y html"<input type=range>"

# ╔═╡ fbf09000-632d-419a-83b8-c2a739c13cdd
y
```
image

`is_first_value` is now inferred by the server, looking if the client sends an `add` patch or an `update` patch. Which is kinda hacky actually (I mention #275 there again) because it only sends an `add` patch when the bond wasn't set by anyone before (so it will send it on a refresh, because it thinks "x already existed on the bonds state, so it isn't the first value!".
I changed this because sending the `is_first_value` to the server is a bit more tricky. But I changed that back now.

**Comment by dralletje** [comment]
Dangit, I'm too used to Firebasey "mutating" by copying that I totally forgot to test that Bond Value mutation line 😂

**Comment by pankgeorg** [comment]
Hahaha! Hopefully the test will warn if #275 comes back too!

**Comment by pankgeorg** [comment]
Ok the good thing is that the test now passes.
The weird thing is that the value it passes with is 3?! (the value stays stable through refreshes)

**Comment by dralletje** [comment]
Yeah, initially it will run the bonds one-by-one because that how it is implemented on the server, will have a look at that now

**Comment by dralletje** [comment]
I made this waaaaay to big very quickly, so I'll put some smaller parts of this out first 😅

**Comment by pankgeorg** [comment]
> I made this waaaaay to big very quickly, so I'll put some smaller parts of this out first 😅

Let me know what chuncks I can bite off this !

File: src/webserver/Dynamic.jl [hunk 1]
```diff
@@ -168,9 +172,11 @@ A placeholder path. The path elements that it replaced will be given to the func
...
 struct Wildcard end
 
-@enum Changed begin
-    CodeChanged
-    FileChanged
+abstract type Changed end
+struct CodeChanged <: Changed end
+struct FileChanged <: Changed end
+struct BondChanged <: Changed
+    bond_name::Symbol
```
Comment (fonsp) [inline src/webserver/Dynamic.jl:179]: omg this is genius

**Comment by pankgeorg** [comment]
@dralletje I think I fixed the `is_first_value` part - I'm not sure if the part where multiple bonds get updated together works properly.

The test that ensures bond stability through refreshes passes.

Can you take a look? 😇😇

**Comment by fonsp** [comment]
Whoopsa this broke custom inputs like `PlutoUI.Scrubbable`

**Pull request merged**